### PR TITLE
[tests-only][full-ci]Add coding standard php pipeline

### PR DESCRIPTION
**Pull request by SagarGi**
### Description

This PR fixes all the errors that appears with command for checking coding-standard-php (locally) and also adds the `coding-standard-php` pipeline on ocis.

### Related Issue
https://github.com/owncloud/ocis/issues/4503

### CODE REVIEW

**Review by SwikritiT — COMMENTED** [review]
Can you also fix this typo `becase` to `because`, while you're working on this file
https://github.com/owncloud/ocis/blob/8514712027a796d6561652020618c1eb1ea6eea1/.drone.star#L2912
https://github.com/owncloud/ocis/blob/8514712027a796d6561652020618c1eb1ea6eea1/.drone.star#L2934

File: .drone.star [hunk 1]
```diff
@@ -148,6 +148,10 @@ config = {
         "architectures": ["arm", "arm64", "amd64"],
     },
     "litmus": True,
+    "codestyle": True,
+    "branches": [
+        "master",
+    ],
```
Comment (saw-jan) [inline .drone.star:154]: add directly like in other pipelines
```py
"ref": [
    "refs/heads/master",
    "refs/pull/**",
    "refs/tags/**",
],
```
```suggestion
```

File: .drone.star [hunk 2]
```diff
@@ -560,6 +562,100 @@ def uploadScanResults(ctx):
...
+        "commands": [
+            "make vendor-bin-codesniffer",
+        ],
+    }]
+
+def codestyle(ctx):
+    pipelines = []
+
+    if "codestyle" not in config:
+        return pipelines
```
Comment (grgprarup) [inline .drone.star:593]:
```suggestion
        return []
```
IMO this will be much easier to understand what it returns.

File: .drone.star [hunk 3]
```diff
@@ -2008,16 +2104,14 @@ def skipIfUnchanged(ctx, type):
...
     skip = []
-    if type == "acceptance-tests":
+    if type == "acceptance-tests" or type == "e2e-tests" or type == "lint":
         skip = base + unit
     if type == "unit-tests":
         skip = base + acceptance
     if type == "build-binary" or type == "build-docker" or type == "litmus":
         skip = base + unit + acceptance
     if type == "cache":
         skip = base
```
Comment (grgprarup) [inline .drone.star:2114]:
```suggestion
    skip = base
    if type == "acceptance-tests" or type == "e2e-tests" or type == "lint":
        skip += unit
    elif type == "unit-tests":
        skip += acceptance
    elif type == "build-binary" or type == "build-docker" or type == "litmus":
        skip += unit + acceptance
```

File: .drone.star [hunk 4]
```diff
@@ -2008,16 +2104,14 @@ def skipIfUnchanged(ctx, type):
     ]
 
     skip = []
```
Comment (grgprarup) [inline .drone.star:2106]:
```suggestion
```

File: .drone.star [hunk 5]
```diff
@@ -2008,16 +2104,14 @@ def skipIfUnchanged(ctx, type):
...
     if type == "unit-tests":
         skip = base + acceptance
     if type == "build-binary" or type == "build-docker" or type == "litmus":
         skip = base + unit + acceptance
     if type == "cache":
         skip = base
-    if type == "e2e-tests":
-        skip = base + unit
     if len(skip) == 0:
         return []
```
Comment (grgprarup) [inline .drone.star:2116]:
```suggestion
```
The default is set to base. so no need of this line.

File: .drone.star (hunk 3 above)
Comment (grgprarup) [inline .drone.star:2114]:
```suggestion
    if type == "acceptance-tests" or type == "e2e-tests" or type == "lint":
        skip = base + unit
    elif type == "unit-tests":
        skip = base + acceptance
    elif type == "build-binary" or type == "build-docker" or type == "litmus":
        skip = base + unit + acceptance
    elif type == "cache":
        skip = base
    else:
        return []
```

File: .drone.star (hunk 5 above)
Comment (grgprarup) [inline .drone.star:2116]:
```suggestion
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 6]
```diff
@@ -3000,9 +3017,15 @@ public function userHasStoredEtagOfElementFromSpace(string $user, string $path,
...
+	 * @param $user
+	 * @param $path
+	 * @param $storePath
+	 * @param $space
+	 *
+	 * @return void
 	 * @throws Exception | GuzzleException
 	 */
-	public function userHasStoredEtagOfElementOnPathFromSpace($user, $path, $storePath, $space) {
+	public function userHasStoredEtagOfElementOnPathFromSpace($user, $path, $storePath, $space):void {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/SpacesContext.php:3028]: Note: we can also declare parameter types. That is done in most of the functions here, but is missing from this function.

@SagarGi please get someone to look through the PHP test code and add type declarations wherever we can.

**Review by phil-davis — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Review by grgprarup — APPROVED** [review]
lgtm :+1:

**Pull request merged**